plugin/evm/header/block_gas_cost.go

```diff
@@ -117,3 +120,70 @@ func EstimateRequiredTip(
 	estimatedTip := totalRequiredTips.Div(totalRequiredTips, totalGasUsed)
 	return estimatedTip, nil
 }
+
+func VerifyBlockFee(
+	baseFee *big.Int,
+	requiredBlockGasCost *big.Int,
+	txs []*types.Transaction,
+	receipts []*types.Receipt,
+	extraStateChangeContribution *big.Int,
+) error {
+	if baseFee == nil || baseFee.Sign() <= 0 {
+		return fmt.Errorf("invalid base fee (%d) in apricot phase 4", baseFee)
+	}
+	if requiredBlockGasCost == nil || !requiredBlockGasCost.IsUint64() {
+		return fmt.Errorf("invalid block gas cost (%d) in apricot phase 4", requiredBlockGasCost)
+	}
```

Comment:
it's not in scope of this PR but we should probably skip verifying block fee if required block gas cost is 0, like here https://github.com/ava-labs/coreth/pull/1215

Reply:
This will be synced when the sync work catches up - right now I am waiting for the existing sync PRs open to be reviewed and merged, as they need to be done in chronological order.